app/src/main/java/io/intrepid/contest/screens/entrysubmission/entryimage/EntryImageActivity.java

```diff
@@ -186,8 +214,24 @@ public void showContestStatusScreen() {
         startActivity(ContestStatusActivity.makeIntent(this));
     }
 
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    @Override
+    public void checkStoragePermissions() {
+        boolean hasPermissions = checkSelfPermission(READ_EXTERNAL_STORAGE) == PERMISSION_GRANTED;
+        Timber.d("Permissions needed " + hasPermissions);
+        presenter.onStoragePermissionCheck(hasPermissions);
+        presenter.onStoragePermissionCheck(hasPermissions);
+    }
+
+    @SuppressLint("NewApi")
```

Comment:
Lint complains that the requestPermissions() method is available only from api 23 and up, whereas our declared api min for the project is 17.